### PR TITLE
End of round cleanup

### PR DESCRIPTION
**Pull request by 11qats**
The ui could use a lot of work, but I have no idea what I'm doing there so I'll leave that to someone else.

I pretty much exclusively play base game so someone with better knowledge of pok should double check I haven't over looked anything it introduces to the end of the round cleanup.

Doesn't check if  a player has technologies that affect the number of command tokens/action cards to receive at the end of the status phase.

I made it into two classes to match place-tradegoods-unpicked.js because I think that file occupies a similar role in the game, but I'm not sure I'm organizing the classes properly.

Open to any and all feedback.

### CODE REVIEW

File: src/lib/phase/end-of-round.js [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+                    commandTokenBag.takeAt(0, dropPosition, true);
+                }
+            }
+        }
+    }
+    /**
+     * Returns all strategy cards to their proper position and rotation.
+     */
+    static returnStrategyCards() {
+        for (const obj of world.getAllObjects()) {
```
Comment (darrellanderson) [inline src/lib/phase/end-of-round.js:275]: out of paranoia, skip if `obj.getContainer()`

File: src/lib/phase/end-of-round.js [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+     * and relics.
+     */
+    static refreshCards() {
+        for (const obj of world.getAllObjects()) {
+            if (obj.getContainer()) {
+                continue; // ignore objects in containers
+            }
+
+            if (ObjectNamespace.isCard(obj)) {
+                if (obj.getStackSize() > 1) {
```
Comment (darrellanderson) [inline src/lib/phase/end-of-round.js:312]: CardUtil.isLooseCard has this plus a number of other checks, you might want to use that

File: src/lib/phase/end-of-round.js [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+            dealNTokens++;
+        }
+
+        return dealNTokens;
+    }
+    /**
+     * Repairs all ships that are on system tiles
+     */
+    static repairShips() {
+        for (const obj of world.getAllObjects()) {
```
Comment (darrellanderson) [inline src/lib/phase/end-of-round.js:171]: skip if obj.getContainer()

File: src/setup/setup-strategy-cards.js [hunk 4]
```diff
@@ -95,4 +95,6 @@ class SetupStrategyCards extends AbstractSetup {
     }
 }
 
-module.exports = { SetupStrategyCards };
+// TODO: STRATEGY_CARDS exported for end of round cleanup remove it once
+// strategy cards have a better defined home/mat.
+module.exports = { SetupStrategyCards, STRATEGY_CARDS };
```
Comment (darrellanderson) [inline src/setup/setup-strategy-cards.js:100]: We should probably export a function to get the position/rotation b/c that's going to change when the strategy card mat is added.

**Review by darrellanderson — COMMENTED** [review]
Looks good, a few comments inline

File: src/setup/setup-strategy-cards.js (hunk 4 above)
Comment (11qats) [reply src/setup/setup-strategy-cards.js:100]: I made the other changes, but I think I will move this to a different task/pull request

**Review by darrellanderson — APPROVED** [review]

**Pull request merged**